### PR TITLE
Private-Events Projects

### PR DESCRIPTION
**Pull request by emiliocm9**
### This project had the aim to build a private social club which allows the client to interact with the server to create new users, new events and decide the attendance of every event created. This projects has the next features:

- [x] The client can create new users.
- [x] Every user has the opportunity to create new events and set an attendance list.
- [x] Every user has the opportunity to decide the events to attend.
- [x] The events are separated in two different lists: upcoming events and previous events.
- [x] All users have to have an username and a password to get accepted in the club.
- [x] The app has been developed on Heroku.
- [x] Bootstrap has been used to style the app layout.

### CODE REVIEW

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: %i[show edit update destroy]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+    # @upcoming_events = current_user.upcoming_events
+    # @prev_events = current_user.previous_events
```
Comment (mosesogwo) [inline app/controllers/events_controller.rb]: Remove commented code

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# class to create a new table attendances
+class SessionsController < ApplicationController
+  before_action :authorized, only: [:welcome]
+
+  def new
+    # @session = Session.new
```
Comment (mosesogwo) [inline app/controllers/sessions_controller.rb]: Remove commented code

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  # GET /users.json
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  # GET /users/1.json
+  def show
+    @events = Event.all
+    set_user
```
Comment (mosesogwo) [inline app/controllers/users_controller.rb]: _set\_user_ here seem to be unnecessary since you have _current\_user_ for use in your views.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**Hello! :wave:. Good job here :+1:. Let's now make the following fixes:**

:point_right: The user's show page should display events the user has created and events that the user is attending to separated in past and upcoming. Currently, the page shows all events (ones that he didn't create and is not attending shouldn't be shown). Remember to group them into past and upcoming.





:point_right: In line with the above, your UsersController show action should retrieve these three collections for use in the views: _his created events_, _his attended_events (past)_, _attended_events (upcoming)_



:point_right: Let's add navigation links at the top of the page. Users should be able to navigate to create new event, login, logout and visit the home page from any view. The navigation links will be appropriate in Application Layout.





_**Kindly make these changes and submit another code review request.**_

**Any questions? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes Requested.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
### General


##### The Users `show.html.erb` views should display the following:

* [x] lists events created by user, i.e. `@created_events = current_user.events` 



* [x] lists events previoulsy attendented by user
* [x] lists upcoming events previously attended by user 




### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by Ispirett — APPROVED** [review]
## Status Approved :+1: 

Great job with this project and the changes requested :+1:  :1st_place_medal: you can move along.

**Pull request merged**